sklearn/decomposition/tests/test_pca.py

```diff
@@ -307,7 +307,7 @@ def test_singular_values():
     rpca.fit(X_hat)
     assert_array_almost_equal(pca.singular_values_, [3.142, 2.718, 1.0], 14)
     assert_array_almost_equal(apca.singular_values_, [3.142, 2.718, 1.0], 14)
-    assert_array_almost_equal(rpca.singular_values_, [3.142, 2.718, 1.0], 14)
+    assert_array_almost_equal(rpca.singular_values_, [3.142, 2.718, 1.0], 2)
```

Comment:
Randomized methods are typically not as accurate as the other methods. I'm quite amazed that the previous tests even passed. It's possible to get arbitrary precision by increasing the number of power iterations via the `iterated_power` parameter. Setting `iterated_power=40` in this case fixes the problem, but makes the tests run longer.

I think it's quite sufficient to test up to 2 decimals.

Reply:
Actually, you're completely right. The error should not be this large.

Upon further inspection, I've noticed that the way we currently deal with wide matrices i.e. more features than samples is to simply transpose it and hope for the best. The current implementation doesn't account for this and was, in fact, wrong. The mistake occurred because I was computing column-wise means, which is wrong if the matrix was previously transposed.

I will fix this and ensure good accuracy.